python/ray/data/dataset.py

```diff
@@ -3566,6 +3568,78 @@ def _divide(self, block_idx: int) -> ("Dataset[T]", "Dataset[T]"):
         )
         return l_ds, r_ds
 
+    def default_batch_format(self) -> Type:
+        """Return this dataset's default batch format.
+
+        The default batch format describes what batches of data look like. To learn more
+        about batch formats, read
+        :ref:`writing user-defined functions <transform_datasets_writing_udfs>`.
+
+        Example:
+
+            If your dataset represents a list of Python objects, then the default batch
+            format is ``list``.
+
+            >>> ds = ray.data.range(100)
+            >>> ds  # doctest: +SKIP
+            Dataset(num_blocks=20, num_rows=100, schema=<class 'int'>)
+            >>> ds.default_batch_format()
+            <class 'list'>
+            >>> next(ds.iter_batches(batch_size=4))
+            [0, 1, 2, 3]
+
+            If your dataset contains a single column of ``TensorDtype`` or
+            ``ArrowTensorType``, then the default batch format is ``ndarray``.
+
+            >>> ds = ray.data.range_tensor(100)
+            >>> ds  # doctest: +SKIP
+            Dataset(num_blocks=20, num_rows=100, schema={__value__: ArrowTensorType(shape=(1,), dtype=int64)})
+            >>> ds.default_batch_format()
+            <class 'numpy.ndarray'>
+            >>> next(ds.iter_batches(batch_size=4))
+            array([[0],
+                   [1],
+                   [2],
+                   [3]])
+
+            If your dataset represents structured data and the default batch format
+            isn't ``ndarray``, then the default batch format is ``DataFrame``.
```

Comment:
```suggestion
            If your dataset represents tabular (structured, multi-column) data, then the
            default batch format is ``pd.DataFrame``.
```

Reply:
The dataset doesn't need to be multi-column, right?

```
>>> df = pd.DataFrame({"X": [0, 0, 0]})
>>> ds = ray.data.from_pandas(df)
>>> type(next(ds.iter_batches()))
<class 'pandas.core.frame.DataFrame'>
```

Reply:
Is this okay?

```
If your dataset represents tabular data and the default batch format
isn't ``np.ndarray``, then the default batch format is ``pd.DataFrame``.
```

Reply:
Yeah you're right, "multi-column" is the common delineation between tabular vs. tensor datasets, but it's not technically correct. Still, we should try to make the "If A, then default format B" declaration self-contained for each of these so users don't have to hop between them. 🤔 

How about
```
If your dataset represents tabular data and doesn't only consist of a `__value__` tensor column (such as is created by ``ray.data.from_numpy()``), then the default batch format is ``pd.DataFrame``.
```

Reply:
Yeah, makes sense. Addressed.